crates/transaction-pool/src/test_utils/mock.rs

```diff
@@ -1156,7 +1160,27 @@ impl MockTransactionSet {
             txs.push(curr_tx.clone());
         }
 
-        MockTransactionSet::new(txs)
+        Self::new(txs)
+    }
+
+    /// Creates a chain of transactions for a given sender with a specified count.
+    ///
+    /// This method generates a sequence of transactions starting from the specified sender
+    /// and creates a chain of transactions based on the `tx_count`.
+    pub fn sequential_transactions_by_sender(
+        sender: Address,
+        tx_count: usize,
+        tx_type: TxType,
+    ) -> Self {
+        let mut txs = Vec::with_capacity(tx_count);
+        let mut current_tx = MockTransaction::new_from_type(tx_type).with_sender(sender);
+
+        for _ in 0..tx_count {
+            txs.push(current_tx.clone());
+            current_tx = current_tx.next();
+        }
+
+        Self::new(txs)
```

Comment:
this could probably just call `dependent` with a `from_nonce` of 0

Reply:
yes, just done